### PR TITLE
docs(style): XDS code style guide + Calendar as reference implementation

### PR DESCRIPTION
**Pull request by cixzhang**
## Summary

Establishes XDS code style standards with Calendar as the reference implementation.

### The Standards

**File order within a component file:**

1. Imports
2. Types/interfaces (prop types live here for easy introspection)
3. Component function (render logic — the main content)
4. Sub-components (private)
5. Styles (StyleX at the bottom)

**Folder structure:**

```
ComponentName/
├── index.ts              # Public API exports
├── XDSComponentName.tsx  # Component: types, render, sub-components, styles
├── types.ts              # Shared/composition types (NOT prop types)
├── hooks/                # Public API hooks (own files, own tests)
├── hooks.ts              # Internal component hooks (when extracted)
├── utils.ts              # Pure utility functions
├── XDSComponentName.test.tsx
└── utils.test.ts
```

**Key decisions:**
- Prop types stay in the component file (not extracted to types.ts)
- Styles stay in the component file at the bottom (no separate styles.ts)
- Internal hooks vs public API hooks are distinct — public hooks get own files
- Complex hooks/utils get their own tests
- SYNC comments are preserved to reduce drift
- Comments should be short, useful, and explain *why* not *what*

### Calendar refactor (reference impl)

- Inlined styles from `styles.ts` into `XDSCalendar.tsx` (bottom)
- Prop types remain in component file
- `types.ts` reduced to shared re-exports only
- Preserves the Calendar SYNC block, including CLI showcase block references
- Keeps Calendar on `xdsThemeProps` after rebasing onto current `main`
- Added `useCalendarDays.test.ts` and `useCalendarConstraints.test.ts` (11 hook tests)

### Enforcement

- Existing: `eslint-plugin-xds` rules (boolean naming, copyright, docblock format)
- Planned: comment verbosity lint, file order lint
- Manual: code review against the guide

### Verification
- [x] `pnpm check:repo`
- [x] `pnpm lint`
- [x] `pnpm -F @xds/core typecheck`
- [x] `pnpm test packages/core/src/Calendar/hooks/useCalendarConstraints.test.ts packages/core/src/Calendar/hooks/useCalendarDays.test.ts`

---

*Draft — reviewing standards before applying across the codebase.*

### CODE REVIEW

File: docs/CODE_STYLE.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+### Component folder structure
+
+```
+ComponentName/
+├── index.ts              # Public API exports
+├── XDSComponentName.tsx  # Main component file
+├── types.ts              # Component-specific types (when shared across files)
+├── styles.ts             # StyleX styles
+├── hooks.ts (or hooks/)  # Component-specific hooks
```
Comment (cixzhang) [inline docs/CODE_STYLE.md]: Specifically, these are hooks used by the component internally. We also have hooks we expose as public APIs which should be their own files.

File: docs/CODE_STYLE.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+## File Organization
+
+### Component folder structure
+
+```
+ComponentName/
+├── index.ts              # Public API exports
+├── XDSComponentName.tsx  # Main component file
+├── types.ts              # Component-specific types (when shared across files)
+├── styles.ts             # StyleX styles
```
Comment (cixzhang) [inline docs/CODE_STYLE.md]: Thinking whether to keep this in the main component file or not. By moving it out we separate the styling concern but if the LLM is reaching for source code to understand visual issues it could result in an additional jump

File: docs/CODE_STYLE.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+## File Organization
+
+### Component folder structure
+
+```
+ComponentName/
+├── index.ts              # Public API exports
+├── XDSComponentName.tsx  # Main component file
+├── types.ts              # Component-specific types (when shared across files)
```
Comment (cixzhang) [inline docs/CODE_STYLE.md]: Prefer keeping prop types in the component file for ease of prop introspection but we can use this to keep other types or compositions.

File: docs/CODE_STYLE.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+### Within a component file
+
+Order within a `.tsx` file:
+
+1. **Imports**
+2. **Types/interfaces** (component props, internal types)
+3. **Component function** — render logic is the main content
+4. **Sub-components** (private, not exported)
```
Comment (cixzhang) [inline docs/CODE_STYLE.md:28]: 5 would be styles if we keep it in this file

File: docs/CODE_STYLE.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+### When to add separate hook/util tests
+
+If it's sufficiently complex — meaning it has branching logic, edge cases, or would be hard to test exhaustively through the component alone. The `dayCellUtils.test.ts` and `useCalendarDays.test.ts` are good examples: they cover edge cases (month boundaries, variable row counts, constraint combinations) that would be cumbersome to test only via the full component.
+
+## Comments
+
+- **Short and useful.** Write for the next person who has to maintain this.
+- **No narration.** Don't describe what the code obviously does (`// increment counter`).
+- **Explain why, not what.** If the code is non-obvious, explain the reasoning.
+- **No LLM verbosity.** Avoid multi-paragraph docstrings for simple functions. One line is fine.
```
Comment (cixzhang) [inline docs/CODE_STYLE.md]: We should preserve the SYNC comments to help with reducing drift
